homeassistant/components/aladdin_connect/__init__.py

```diff
@@ -27,10 +27,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         username, password, async_get_clientsession(hass), CLIENT_ID
     )
     try:
-        if not await acc.login():
-            raise ConfigEntryAuthFailed("Incorrect Password")
+        await acc.login()
     except (ClientConnectionError, asyncio.TimeoutError) as ex:
         raise ConfigEntryNotReady("Can not connect to host") from ex
+    except ClientError as ex:
```

Comment:
You should be able to check the status code.

if status == 401: auth failed, else cannot connect

Reply:
I check the status code in the API, raise the ClientError when it's 401, ClientConnectionError for everything else.  

The API doesn't pass the status back out here (right?)  The API was only passing True/False for a "good" login, but false was set for ANY login failure, including no internet, no connection, etc.  

In my case - power goes out, internet is "slow" to come back, HA always said my password failed :(

Reply:
Paulus is right, `ClientError` is from `aiohttp`, it should not be throwing that as such. Instead catch it correctly in the client library and throw a specific library error that describes the issue. E.g., `InvalidPasswordError` or something similar.

Reply:
Ahh the more pythonic way :)
I believe the commit has addressed  your comments.